rust/osm2lanes/src/transform/tags_to_lanes/lane_count.rs

```diff
@@ -0,0 +1,366 @@
+                            lanes: Infer::Direct(l),
+                            forward: Infer::Direct(half),
+                            backward: Infer::Direct(remaining_lanes.checked_sub(half).unwrap()),
+                            both_ways,
+                        }
+                    }
+                },
+                (None, _, _) => {
+                    // Without the "lanes" tag, assume one normal lane in each dir
+                    // Adding any bus:lanes as needed.
+                    const DEFAULT: usize = 1;
+                    let tagged_bus_lanes: Option<usize> =
+                        tags.get_parsed(&(LANES + "bus"), warnings);
+                    let tagged_bus_forward = tags
+                        .get_parsed(&(LANES + "bus" + "forward"), warnings)
+                        .or_else(|| tagged_bus_lanes.map(|l| l / 2))
+                        .unwrap_or(0);
+                    let tagged_bus_backward = tags
+                        .get_parsed(&(LANES + "bus" + "backward"), warnings)
+                        .or_else(|| tagged_bus_lanes.map(|l| l / 2))
+                        .unwrap_or(0);
+                    let forward = match tagged_forward {
+                        Some(f) => Infer::Direct(f),
+                        None => Infer::Calculated(tagged_bus_forward + DEFAULT),
+                    };
+                    let backward = match tagged_backward {
+                        Some(b) => Infer::Direct(b),
+                        None => Infer::Default(tagged_bus_backward + DEFAULT),
+                    };
+                    let lanes = Infer::Default(
+                        tagged_forward.unwrap_or(DEFAULT)
+                            + tagged_backward.unwrap_or(DEFAULT)
+                            + tagged_bus_lanes.unwrap_or_else(|| {
+                                tagged_bus_forward.checked_add(tagged_bus_backward).unwrap()
+                            })
+                            + both_way_lanes,
+                    );
+                    Self {
+                        lanes,
+                        forward,
+                        backward,
+                        both_ways,
+                    }
+                },
+            }
+        };
+
+        assert!(counts.forward.some().unwrap_or(0) >= pre_forward);
+        assert!(counts.backward.some().unwrap_or(0) >= pre_backward);
+
+        counts
+    }
+
+    /// Look at tags that use pipe separated access values to determine lane counts
+    fn get_counts_by_access(tags: &Tags, warnings: &mut RoadWarnings) -> Self {
+        const LANES: [TagKey; 2] = [TagKey::from("bus:lanes"), TagKey::from("psv:lanes")];
+        const FORWARD: [TagKey; 2] = [
+            TagKey::from("bus:lanes:forward"),
+            TagKey::from("psv:lanes:forward"),
+        ];
+        const BACKWARD: [TagKey; 2] = [
+            TagKey::from("bus:lanes:backward"),
+            TagKey::from("psv:lanes:backward"),
+        ];
```

Comment:
Every instance of `<tag>:lanes` and `<tag>:lanes:{forward,backward}` should be checked, according to https://wiki.openstreetmap.org/wiki/Lanes.

If this calculation remains as part of `lanes` scheme parsing, then I would expect it to look at `*:lanes`, rather than a specific list of tags.

Reply:
oh true, I didn't think of doing `*:lanes` globbed parsing. we don't have a facility to do that so that can be easily added later.